Adding songs for in-simulator testing

johanngan opened the pull request:
Also improvements/fixes in song shuffling

johanngan committed:
Added testing functionality on iOS simulator.

- Included the directory "testSongs/" for simulator testing
- Added DB helper functions addSongToDB and wipeDB to LoopMusicViewController class
- Added suggested (but untested) change to (void)mediaPicker in SettingsViewController as a comment

johanngan committed:
Fixed OBOE in song shuffling procedure, and changed to a less biased RNG. Also made the README in testSongs/ be ignored when reading in songs.

AnonymousRandomPerson commented on LoopMusicViewController.m:
```diff
@@ -115,6 +115,56 @@ - (void)viewDidLoad
+// Load local test files from the testSongs/ directory if testing on the simulator
+#if TARGET_OS_SIMULATOR
+    NSString *localDir = @"testSongs/";
+    if(![self isSongListEmpty])
```
Please follow the coding style of this repository. Space between "if" and parentheses.

johanngan committed:
Added a space between "if"/"for" and parentheses

johanngan committed:
Merge branch 'master' of https://github.com/AnonymousRandomPerson/LoopMusic

johanngan committed:
Changed getSongIndices not to return NULL for the "All tracks" playlist, and fixed shuffling accordingly. Also used the less biased RNG for timeVariance calculation

AnonymousRandomPerson commented on LoopMusicViewController.m:
```diff
@@ -893,13 +886,31 @@ - (NSArray*)getSongIndices
-    [self prepareQuery:[NSString stringWithFormat:@"SELECT tracks FROM Playlists WHERE id = %ld", (long)playlistIndex]];
-    if (sqlite3_step(statement) == SQLITE_ROW)
+    NSString *trackString = nil;
+    if(playlistIndex)
```
Space before parentheses.

AnonymousRandomPerson commented on LoopMusicViewController.m:
```diff
@@ -893,13 +886,31 @@ - (NSArray*)getSongIndices
+        {
+            /// The database string containing the current track ID
+            NSString *idString = [[NSString alloc] initWithUTF8String:(const char *) sqlite3_column_text(statement, 0)];
+            trackString = [[trackString stringByAppendingString:idString] stringByAppendingString:@","];
```
This concatenates all IDs together as a string before splitting that string back up again. Can the IDs be immediately added to an array as they are processed?

johanngan replied:
I couldn't figure out how to do it... As far as I can tell you can't initialize a blank NSArray, and a mutable array doesn't work here because of the function return value. Suggestions?

AnonymousRandomPerson replied:
The copy function can return an immutable NSArray from a NSMutableArray.

AnonymousRandomPerson commented on LoopMusicViewController.h:
```diff
@@ -32,6 +32,8 @@ extern NSInteger playlistIndex;
     /// The database index of the current track.
     NSInteger musicNumber;
+    /// The index of the current track within the current playlist
+    NSInteger musicIndex;
```
This isn't used anywhere.

johanngan committed:
Changed getSongIndices to use NSMutableArray instead of strings for the All tracks playlist. Made whitespace style consistent. Fixed the simulator not loading if only one track is in the testSongs directory.

johanngan committed:
Changed Playlists table structure.

-Playlists has columns: (rowkey integer PRIMARY KEY, id integer, track integer)
    -Each track in a playlist has its own corresponding row with shared (playlist) id value.
-Added new table called PlaylistNames with columns: (id integer PRIMARY KEY, name text). Each playlist is listed once.
-Updated functions that used the old Playlists format.
-Changed simulator song loading message to use 1-indexing.

johanngan committed:
Flagged possibly problematic line.
